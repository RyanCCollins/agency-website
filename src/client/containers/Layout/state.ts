
export interface State {
  isMobile: boolean;
  brandText: string;
};

export const initialState: State = {
  isMobile: false,
  brandText: 'The Agency',
};

export default State;
