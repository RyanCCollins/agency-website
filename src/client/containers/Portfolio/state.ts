import { ErrorType } from './types';

export interface Project {
  title: string;
  image: string;
}

export interface Data {
  items: Project[],
}

export interface State {
  isLoading: boolean;
  error?: ErrorType;
  data?: Data;
}

export const initialState: State = {
  isLoading: false,
  error: null,
  data: {
    items: [
      {
        title: 'HPE Brand Central',
        image: 'https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmain.png?raw=true',
      },
      {
        title: 'Grommet CMS',
        image: 'https://github.com/RyanCCollins/cdn/blob/master/cms/main.png?raw=true',
      },
      {
        title: 'HPE Datacenter Care',
        image: 'https://github.com/RyanCCollins/cdn/blob/master/datacenter-care/main.png?raw=true',
      },
      {
        title: 'Astra Finance',
        image: 'https://github.com/RyanCCollins/cdn/blob/master/astra.jpg?raw=true',
      },
      {
        title: 'Udacity Mentor Calendar',
        image: 'https://github.com/RyanCCollins/cdn/blob/master/'
          + 'portfolio-image-gallery-images/mentor-calendar/main-mockup.jpg?raw=true',
      },
    ],
  },
};

export default initialState;
