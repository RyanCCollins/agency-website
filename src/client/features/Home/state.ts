import { ErrorType } from './types';

export interface Button {
  label: string;
  path: string;
  color: string;
}

export interface Section {
  text: string;
  color: string;
  button?: Button;
  isVisible: boolean;
}

export interface State {
  isLoading: boolean;
  error?: ErrorType;
  data?: string;
  sections: Section[];
}

export const initialState: State = {
  isLoading: false,
  error: null,
  data: null,
  sections: [
    {
      text: 'We are a dedicated digital agency',
      color: 'secondary',
      isVisible: false,
    },
    {
      text: 'We build scalable solutions for the web and mobile platforms',
      color: 'dark1',
      button: {
        color: 'white1',
        path: '/case-studies',
        label: 'See Case Studies',
      },
      isVisible: false,
    },
    {
      text: 'Need our help?',
      color: 'light2',
      button: {
        color: 'black1',
        path: '/contact',
        label: 'Get in touch',
      },
      isVisible: false,
    },
  ],
};

export default initialState;
