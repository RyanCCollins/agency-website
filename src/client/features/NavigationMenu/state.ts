export interface NavLink {
  path: string;
  label: string;
}

export interface State {
  isVisible: boolean;
  navLinks: NavLink[];
}

export const initialState: State = {
  isVisible: false,
  navLinks: [
    {
      path: '/case-studies',
      label: 'Case Studies',
    },
    {
      path: '/clients',
      label: 'Clients',
    },
    {
      path: '/contact',
      label: 'Contact',
    },
  ],
};

export default initialState;
