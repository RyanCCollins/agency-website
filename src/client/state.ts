/* GENERATOR-IMPORT-STATE */
import { initialState as homeState, State as HomeState } from './containers/Home/state';
import { initialState as appState, State as AppState } from 'containers/App/state';

export interface State {
  /* GENERATOR-EXPORT-STATE-TYPE */
  home: HomeState;
  app: AppState;
}

export const initialState: State = {
  /* GENERATOR-EXPORT-STATE */
  home: homeState,
  app: appState,
};
