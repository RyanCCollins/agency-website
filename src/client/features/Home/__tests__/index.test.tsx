import * as React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { mountToJson } from 'enzyme-to-json';
import { Store } from 'redux';
import { Provider } from 'react-redux';
import mockStore from '../../../test/mockstore';
import theme from '../../../theming';
import HomeContainer from '../';
import { State } from '../state';
import Presentation from '../presentation';

const testState: State = {
  isLoading: false,
  data: 'foo',
  error: new Error('Ooops'),
};

describe('HomeContainer ', () => {
  const store: Store<State> = mockStore({ home: testState });

  const wrapper = mount(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <HomeContainer />
      </ThemeProvider>
    </Provider>,
  );

  it('should create a snapshot', () => {
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render a Presentation component', () => {
    expect(wrapper.find(Presentation).length).toBe(1);
  });
});
