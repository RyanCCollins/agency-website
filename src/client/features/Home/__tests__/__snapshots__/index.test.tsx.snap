// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer  should create a snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "black1": "#0a0a0a",
        "black2": "#2d2d2d",
        "black3": "#555555",
        "dark1": "#13293D",
        "dark2": "#006494",
        "dark3": "#92374D",
        "error": "#ff324d",
        "gray1": "#696969",
        "gray2": "#7c7c7c",
        "gray3": "#828282",
        "light1": "#0072BB",
        "light2": "#ff9800",
        "light3": "#0C7489",
        "offwhite": "#f5f5f5",
        "ok": "#8cc800",
        "primary": "#03A9F4",
        "secondary": "#9c27b0",
        "warning": "#ffd602",
        "white1": "#fff",
        "white2": "#f3f3f3",
        "white3": "#e6e8ec",
      }
    }
  >
    <Connect(_class)>
      <_class
        actions={
          Object {
            "loadCancel": [Function],
            "loadFailure": [Function],
            "loadInitiation": [Function],
            "loadSuccess": [Function],
          }
        }
        data="foo"
        error={[Error: Ooops]}
        isLoading={false}
      >
        <Home
          actions={
            Object {
              "loadCancel": [Function],
              "loadFailure": [Function],
              "loadInitiation": [Function],
              "loadSuccess": [Function],
            }
          }
          data="foo"
          error={[Error: Ooops]}
          isLoading={false}
          theme={
            Object {
              "black1": "#0a0a0a",
              "black2": "#2d2d2d",
              "black3": "#555555",
              "dark1": "#13293D",
              "dark2": "#006494",
              "dark3": "#92374D",
              "error": "#ff324d",
              "gray1": "#696969",
              "gray2": "#7c7c7c",
              "gray3": "#828282",
              "light1": "#0072BB",
              "light2": "#ff9800",
              "light3": "#0C7489",
              "offwhite": "#f5f5f5",
              "ok": "#8cc800",
              "primary": "#03A9F4",
              "secondary": "#9c27b0",
              "warning": "#ffd602",
              "white1": "#fff",
              "white2": "#f3f3f3",
              "white3": "#e6e8ec",
            }
          }
        >
          <Presentation
            actions={
              Object {
                "loadCancel": [Function],
                "loadFailure": [Function],
                "loadInitiation": [Function],
                "loadSuccess": [Function],
              }
            }
            data="foo"
            error={[Error: Ooops]}
            isLoading={false}
            sections={
              Array [
                false,
                false,
                false,
              ]
            }
            theme={
              Object {
                "black1": "#0a0a0a",
                "black2": "#2d2d2d",
                "black3": "#555555",
                "dark1": "#13293D",
                "dark2": "#006494",
                "dark3": "#92374D",
                "error": "#ff324d",
                "gray1": "#696969",
                "gray2": "#7c7c7c",
                "gray3": "#828282",
                "light1": "#0072BB",
                "light2": "#ff9800",
                "light3": "#0C7489",
                "offwhite": "#f5f5f5",
                "ok": "#8cc800",
                "primary": "#03A9F4",
                "secondary": "#9c27b0",
                "warning": "#ffd602",
                "white1": "#fff",
                "white2": "#f3f3f3",
                "white3": "#e6e8ec",
              }
            }
          >
            <styled.div>
              <div
                className="jIwUpA"
              >
                <Hero
                  backgroundColor="#03A9F4"
                >
                  <styled.div
                    backgroundColor="#03A9F4"
                    full={false}
                  >
                    <div
                      className="vaLLK"
                    >
                      <Box
                        alignItems="center"
                        justifyContent="center"
                        style={
                          Object {
                            "minHeight": "calc(100vh - 100px)",
                          }
                        }
                      >
                        <styled.div
                          alignItems="center"
                          justifyContent="center"
                          style={
                            Object {
                              "minHeight": "calc(100vh - 100px)",
                            }
                          }
                        >
                          <div
                            className="kpbQte"
                            style={
                              Object {
                                "minHeight": "calc(100vh - 100px)",
                              }
                            }
                          >
                            <styled.img
                              alt="The Agency"
                              size={300}
                              src="https://github.com/RyanCCollins/cdn/blob/master/misc/rocket.png?raw=true"
                            >
                              <img
                                alt="The Agency"
                                className="ddORmD"
                                size={300}
                                src="https://github.com/RyanCCollins/cdn/blob/master/misc/rocket.png?raw=true"
                              />
                            </styled.img>
                            <Headline
                              color="#fff"
                              fontSize="medium"
                              fontWeight={400}
                              margin="medium"
                              textAlign="center"
                            >
                              <styled.h1
                                color="#fff"
                                fontSize="medium"
                                fontWeight={400}
                                margin="medium"
                                textAlign="center"
                              >
                                <h1
                                  className="cSbdEv"
                                  color="#fff"
                                  fontSize="medium"
                                  fontWeight={400}
                                >
                                  The Agency
                                </h1>
                              </styled.h1>
                            </Headline>
                          </div>
                        </styled.div>
                      </Box>
                    </div>
                  </styled.div>
                </Hero>
                <div
                  id="section-one"
                >
                  <Hero
                    backgroundColor="#9c27b0"
                  >
                    <styled.div
                      backgroundColor="#9c27b0"
                      full={false}
                    >
                      <div
                        className="iVIiVF"
                      >
                        <Box
                          alignItems="center"
                          justifyContent="center"
                          style={
                            Object {
                              "minHeight": "calc(100vh - 100px)",
                            }
                          }
                        >
                          <styled.div
                            alignItems="center"
                            justifyContent="center"
                            style={
                              Object {
                                "minHeight": "calc(100vh - 100px)",
                              }
                            }
                          >
                            <div
                              className="kpbQte"
                              style={
                                Object {
                                  "minHeight": "calc(100vh - 100px)",
                                }
                              }
                            >
                              <Box
                                justifyContent="center"
                                style={
                                  Object {
                                    "maxWidth": 576,
                                  }
                                }
                              >
                                <styled.div
                                  justifyContent="center"
                                  style={
                                    Object {
                                      "maxWidth": 576,
                                    }
                                  }
                                >
                                  <div
                                    className="hfaZSB"
                                    style={
                                      Object {
                                        "maxWidth": 576,
                                      }
                                    }
                                  >
                                    <WithAnimation
                                      isVisible={false}
                                      type="fadeInUp"
                                    >
                                      <styled.div
                                        delay={0}
                                        duration={1000}
                                        isVisible={false}
                                        type="fadeInUp"
                                      >
                                        <div
                                          className="eArchH"
                                          type="fadeInUp"
                                        >
                                          <Headline
                                            color="#fff"
                                            fontSize="medium"
                                            fontWeight={400}
                                            margin="medium"
                                            textAlign="center"
                                          >
                                            <styled.h1
                                              color="#fff"
                                              fontSize="medium"
                                              fontWeight={400}
                                              margin="medium"
                                              textAlign="center"
                                            >
                                              <h1
                                                className="cSbdEv"
                                                color="#fff"
                                                fontSize="medium"
                                                fontWeight={400}
                                              >
                                                We are a dedicated digital agency
                                              </h1>
                                            </styled.h1>
                                          </Headline>
                                        </div>
                                      </styled.div>
                                    </WithAnimation>
                                  </div>
                                </styled.div>
                              </Box>
                            </div>
                          </styled.div>
                        </Box>
                      </div>
                    </styled.div>
                  </Hero>
                </div>
                <div
                  id="section-two"
                >
                  <Hero
                    backgroundColor="#0C7489"
                  >
                    <styled.div
                      backgroundColor="#0C7489"
                      full={false}
                    >
                      <div
                        className="jCPwXi"
                      >
                        <Box
                          alignItems="center"
                          justifyContent="center"
                          style={
                            Object {
                              "minHeight": "calc(100vh - 100px)",
                            }
                          }
                        >
                          <styled.div
                            alignItems="center"
                            justifyContent="center"
                            style={
                              Object {
                                "minHeight": "calc(100vh - 100px)",
                              }
                            }
                          >
                            <div
                              className="kpbQte"
                              style={
                                Object {
                                  "minHeight": "calc(100vh - 100px)",
                                }
                              }
                            >
                              <Box
                                justifyContent="center"
                                style={
                                  Object {
                                    "maxWidth": 576,
                                  }
                                }
                              >
                                <styled.div
                                  justifyContent="center"
                                  style={
                                    Object {
                                      "maxWidth": 576,
                                    }
                                  }
                                >
                                  <div
                                    className="hfaZSB"
                                    style={
                                      Object {
                                        "maxWidth": 576,
                                      }
                                    }
                                  >
                                    <WithAnimation
                                      isVisible={false}
                                      type="fadeInUp"
                                    >
                                      <styled.div
                                        delay={0}
                                        duration={1000}
                                        isVisible={false}
                                        type="fadeInUp"
                                      >
                                        <div
                                          className="eArchH"
                                          type="fadeInUp"
                                        >
                                          <Headline
                                            color="#fff"
                                            fontSize="medium"
                                            fontWeight={400}
                                            margin="medium"
                                            textAlign="center"
                                          >
                                            <styled.h1
                                              color="#fff"
                                              fontSize="medium"
                                              fontWeight={400}
                                              margin="medium"
                                              textAlign="center"
                                            >
                                              <h1
                                                className="cSbdEv"
                                                color="#fff"
                                                fontSize="medium"
                                                fontWeight={400}
                                              >
                                                We build scalable solutions for the web and mobile platforms
                                              </h1>
                                            </styled.h1>
                                          </Headline>
                                          <Button
                                            backgroundColor="transparent"
                                            borderColor="#fff"
                                            color="#fefefe"
                                            fontSize="medium"
                                            label="See Case Studies"
                                            path="/case-studies"
                                          >
                                            <withRouter(Anchor)
                                              path="/case-studies"
                                              plain={true}
                                            >
                                              <Anchor
                                                method="push"
                                                path="/case-studies"
                                                plain={true}
                                              >
                                                <styled.a
                                                  onClick={[Function]}
                                                  plain={true}
                                                >
                                                  <a
                                                    className="ckTJUm"
                                                    onClick={[Function]}
                                                  >
                                                    <styled.button
                                                      backgroundColor="transparent"
                                                      borderColor="#fff"
                                                      color="#fefefe"
                                                      fontSize="medium"
                                                    >
                                                      <button
                                                        className="llBEdw"
                                                        color="#fefefe"
                                                        fontSize="medium"
                                                      >
                                                        See Case Studies
                                                      </button>
                                                    </styled.button>
                                                  </a>
                                                </styled.a>
                                              </Anchor>
                                            </withRouter(Anchor)>
                                          </Button>
                                        </div>
                                      </styled.div>
                                    </WithAnimation>
                                  </div>
                                </styled.div>
                              </Box>
                            </div>
                          </styled.div>
                        </Box>
                      </div>
                    </styled.div>
                  </Hero>
                </div>
                <div
                  id="section-three"
                >
                  <Hero
                    backgroundColor="#03A9F4"
                  >
                    <styled.div
                      backgroundColor="#03A9F4"
                      full={false}
                    >
                      <div
                        className="vaLLK"
                      >
                        <Box
                          alignItems="center"
                          justifyContent="center"
                          style={
                            Object {
                              "minHeight": "calc(100vh - 100px)",
                            }
                          }
                        >
                          <styled.div
                            alignItems="center"
                            justifyContent="center"
                            style={
                              Object {
                                "minHeight": "calc(100vh - 100px)",
                              }
                            }
                          >
                            <div
                              className="kpbQte"
                              style={
                                Object {
                                  "minHeight": "calc(100vh - 100px)",
                                }
                              }
                            >
                              <Box
                                justifyContent="center"
                                style={
                                  Object {
                                    "maxWidth": 576,
                                  }
                                }
                              >
                                <styled.div
                                  justifyContent="center"
                                  style={
                                    Object {
                                      "maxWidth": 576,
                                    }
                                  }
                                >
                                  <div
                                    className="hfaZSB"
                                    style={
                                      Object {
                                        "maxWidth": 576,
                                      }
                                    }
                                  >
                                    <WithAnimation
                                      isVisible={false}
                                      type="fadeInUp"
                                    >
                                      <styled.div
                                        delay={0}
                                        duration={1000}
                                        isVisible={false}
                                        type="fadeInUp"
                                      >
                                        <div
                                          className="eArchH"
                                          type="fadeInUp"
                                        >
                                          <Headline
                                            color="#fff"
                                            fontSize="medium"
                                            fontWeight={400}
                                            margin="medium"
                                            textAlign="center"
                                          >
                                            <styled.h1
                                              color="#fff"
                                              fontSize="medium"
                                              fontWeight={400}
                                              margin="medium"
                                              textAlign="center"
                                            >
                                              <h1
                                                className="cSbdEv"
                                                color="#fff"
                                                fontSize="medium"
                                                fontWeight={400}
                                              >
                                                Need our help?
                                              </h1>
                                            </styled.h1>
                                          </Headline>
                                          <Button
                                            backgroundColor="transparent"
                                            borderColor="#fff"
                                            color="#fefefe"
                                            fontSize="medium"
                                            label="Get in Touch"
                                            path="/contact"
                                          >
                                            <withRouter(Anchor)
                                              path="/contact"
                                              plain={true}
                                            >
                                              <Anchor
                                                method="push"
                                                path="/contact"
                                                plain={true}
                                              >
                                                <styled.a
                                                  onClick={[Function]}
                                                  plain={true}
                                                >
                                                  <a
                                                    className="ckTJUm"
                                                    onClick={[Function]}
                                                  >
                                                    <styled.button
                                                      backgroundColor="transparent"
                                                      borderColor="#fff"
                                                      color="#fefefe"
                                                      fontSize="medium"
                                                    >
                                                      <button
                                                        className="llBEdw"
                                                        color="#fefefe"
                                                        fontSize="medium"
                                                      >
                                                        Get in Touch
                                                      </button>
                                                    </styled.button>
                                                  </a>
                                                </styled.a>
                                              </Anchor>
                                            </withRouter(Anchor)>
                                          </Button>
                                        </div>
                                      </styled.div>
                                    </WithAnimation>
                                  </div>
                                </styled.div>
                              </Box>
                            </div>
                          </styled.div>
                        </Box>
                      </div>
                    </styled.div>
                  </Hero>
                </div>
              </div>
            </styled.div>
          </Presentation>
        </Home>
      </_class>
    </Connect(_class)>
  </ThemeProvider>
</Provider>
`;
